src/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs

```diff
@@ -53,21 +53,64 @@ public static SecurityStatusPal InitializeSecurityContext(SafeFreeCredentials cr
             return new SafeFreeSslCredentials(certificate, protocols, policy);
         }
 
-        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] input, int offset, int size, int headerSize, int trailerSize, ref byte[] output, out int resultSize)
+        public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, ReadOnlyMemory<byte> input, int headerSize, int trailerSize, ref byte[] output, out int resultSize)
         {
-            return EncryptDecryptHelper(securityContext, input, offset, size, true, ref output, out resultSize);
```

Comment:
Why did EncryptDecryptHelper get manually inlined?

Reply:
Because encrypt and decrypt are fundamentally different now (actually always were) as the input and output have to be the same for decrypt... it's mutable and for encrypt it's explicitly immutable. It always seemed odd they were combined in openssl but nothing else.

Reply:
So now you would need an extrad read-onlymemory field just for encrypt... which seems odd. I can factor out the case if u want that common. But even then it's not 100% the same.

Reply:
If it makes sense to keep it factored out, I prefer minimizing duplication.  But if it's problematic, separate is fine.

Reply:
Take a look at the last commit (I didn't comment as it was on the train to work and ran out of time). But it puts it back. See what you think but it does seem a little odd now. 

My thoughts, leave it as the last commit, both together and when the decrypt side is tackled it might make sense to split then.

Reply:
(I will remove the sln changes :) )

Reply:
The duplication/shared code is basically the errorcode switch; could  just factor that out?
```csharp
static SecurityStatusPal EncryptDecryptResult(Interop.Ssl.SslErrorCode errorCode)
{
    switch (errorCode)
    {
        case Interop.Ssl.SslErrorCode.SSL_ERROR_RENEGOTIATE:
            return new SecurityStatusPal(SecurityStatusPalErrorCode.Renegotiate);
        case Interop.Ssl.SslErrorCode.SSL_ERROR_ZERO_RETURN:
            return new SecurityStatusPal(SecurityStatusPalErrorCode.ContextExpired);
        case Interop.Ssl.SslErrorCode.SSL_ERROR_NONE:
        case Interop.Ssl.SslErrorCode.SSL_ERROR_WANT_READ:
            return new SecurityStatusPal(SecurityStatusPalErrorCode.OK);
        default:
            return new SecurityStatusPal(SecurityStatusPalErrorCode.InternalError, new Interop.OpenSsl.SslException((int)errorCode));
    }
}
```

Reply:
@benaadams You okay with leaving it as is for now (although its uncomfortable for me to have different paths for encrypt/decrypt all the way to this method then combine and branch). I will not leave it as is in some future PR I promise but is better to do when the read side is in  a state to get memory etc.